<p style="text-align:center;font-size:30px ;font-weight: bold; line-height: 40px;">  HEALTHCARE PROVIDER FRAUD DETECTION ANALYSIS<p/>
<p style="text-align:center; font-size:15px;  font-weight: bold;"> Report about missing values (percentage, count,..) & features data type <p/>
<p style="text-align:center;  line-height: 0px;font-weight: bold;"> July 29, 2021<p/>



# ` Importing libraries`

In [1]:
import pandas as pd 

# `Reading the datasets`

In [2]:
inpatient_df = pd.read_csv("Train_Inpatientdata.csv")
outpatient_df = pd.read_csv('Train_Outpatientdata.csv')

In [3]:
# view first few records
inpatient_df.head()

BeneID   ClaimID ClaimStartDt  ClaimEndDt  Provider  \
0  BENE11001  CLM46614   2009-04-12  2009-04-18  PRV55912   
1  BENE11001  CLM66048   2009-08-31  2009-09-02  PRV55907   
2  BENE11001  CLM68358   2009-09-17  2009-09-20  PRV56046   
3  BENE11011  CLM38412   2009-02-14  2009-02-22  PRV52405   
4  BENE11014  CLM63689   2009-08-13  2009-08-30  PRV56614   

   InscClaimAmtReimbursed AttendingPhysician OperatingPhysician  \
0                   26000          PHY390922                NaN   
1                    5000          PHY318495          PHY318495   
2                    5000          PHY372395                NaN   
3                    5000          PHY369659          PHY392961   
4                   10000          PHY379376          PHY398258   

  OtherPhysician AdmissionDt  ... ClmDiagnosisCode_7  ClmDiagnosisCode_8  \
0            NaN  2009-04-12  ...               2724               19889   
1            NaN  2009-08-31  ...                NaN                 NaN   
2      PHY324689  2009-09-17  ...                NaN                 NaN   
3      PHY349768  2009-02-14  ...              25062               40390   
4            NaN  2009-08-13  ...               5119               29620   

  ClmDiagnosisCode_9 ClmDiagnosisCode_10 ClmProcedureCode_1  \
0               5849                 NaN                NaN   
1                NaN                 NaN             7092.0   
2                NaN                 NaN                NaN   
3               4019                 NaN              331.0   
4              20300                 NaN             3893.0   

  ClmProcedureCode_2 ClmProcedureCode_3 ClmProcedureCode_4 ClmProcedureCode_5  \
0                NaN                NaN                NaN                NaN   
1                NaN                NaN                NaN                NaN   
2                NaN                NaN                NaN                NaN   
3                NaN                NaN                NaN                NaN   
4                NaN                NaN                NaN                NaN   

  ClmProcedureCode_6  
0                NaN  
1                NaN  
2                NaN  
3                NaN  
4                NaN  

[5 rows x 30 columns]

In [4]:
# view first few records
outpatient_df.head()

BeneID    ClaimID ClaimStartDt  ClaimEndDt  Provider  \
0  BENE11002  CLM624349   2009-10-11  2009-10-11  PRV56011   
1  BENE11003  CLM189947   2009-02-12  2009-02-12  PRV57610   
2  BENE11003  CLM438021   2009-06-27  2009-06-27  PRV57595   
3  BENE11004  CLM121801   2009-01-06  2009-01-06  PRV56011   
4  BENE11004  CLM150998   2009-01-22  2009-01-22  PRV56011   

   InscClaimAmtReimbursed AttendingPhysician OperatingPhysician  \
0                      30          PHY326117                NaN   
1                      80          PHY362868                NaN   
2                      10          PHY328821                NaN   
3                      40          PHY334319                NaN   
4                     200          PHY403831                NaN   

  OtherPhysician ClmDiagnosisCode_1  ... ClmDiagnosisCode_9  \
0            NaN              78943  ...                NaN   
1            NaN               6115  ...                NaN   
2            NaN               2723  ...                NaN   
3            NaN              71988  ...                NaN   
4            NaN              82382  ...                NaN   

  ClmDiagnosisCode_10 ClmProcedureCode_1 ClmProcedureCode_2  \
0                 NaN                NaN                NaN   
1                 NaN                NaN                NaN   
2                 NaN                NaN                NaN   
3                 NaN                NaN                NaN   
4                 NaN                NaN                NaN   

  ClmProcedureCode_3 ClmProcedureCode_4 ClmProcedureCode_5 ClmProcedureCode_6  \
0                NaN                NaN                NaN                NaN   
1                NaN                NaN                NaN                NaN   
2                NaN                NaN                NaN                NaN   
3                NaN                NaN                NaN                NaN   
4                NaN                NaN                NaN                NaN   

  DeductibleAmtPaid  ClmAdmitDiagnosisCode  
0                 0                  56409  
1                 0                  79380  
2                 0                    NaN  
3                 0                    NaN  
4                 0                  71947  

[5 rows x 27 columns]

***
# `Data overview`

In [5]:
#get summary of an inpatient dataframe
inpatient_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40474 entries, 0 to 40473
Data columns (total 30 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   BeneID                  40474 non-null  object 
 1   ClaimID                 40474 non-null  object 
 2   ClaimStartDt            40474 non-null  object 
 3   ClaimEndDt              40474 non-null  object 
 4   Provider                40474 non-null  object 
 5   InscClaimAmtReimbursed  40474 non-null  int64  
 6   AttendingPhysician      40362 non-null  object 
 7   OperatingPhysician      23830 non-null  object 
 8   OtherPhysician          4690 non-null   object 
 9   AdmissionDt             40474 non-null  object 
 10  ClmAdmitDiagnosisCode   40474 non-null  object 
 11  DeductibleAmtPaid       39575 non-null  float64
 12  DischargeDt             40474 non-null  object 
 13  DiagnosisGroupCode      40474 non-null  object 
 14  ClmDiagnosisCode_1      40474 non-null

<div class="alert alert-block alert-info"> <strong>Inpatient</strong> dataframe consists of 30 columns(features)
</div>


<p class="alert alert-block alert-warning"> Inpatient dataframe consists of 3 extra columns Admission date, Discharge date, and Diagnosis Group code.

In [6]:
#get summary of an outpatient dataframe
outpatient_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 517737 entries, 0 to 517736
Data columns (total 27 columns):
 #   Column                  Non-Null Count   Dtype  
---  ------                  --------------   -----  
 0   BeneID                  517737 non-null  object 
 1   ClaimID                 517737 non-null  object 
 2   ClaimStartDt            517737 non-null  object 
 3   ClaimEndDt              517737 non-null  object 
 4   Provider                517737 non-null  object 
 5   InscClaimAmtReimbursed  517737 non-null  int64  
 6   AttendingPhysician      516341 non-null  object 
 7   OperatingPhysician      90617 non-null   object 
 8   OtherPhysician          195046 non-null  object 
 9   ClmDiagnosisCode_1      507284 non-null  object 
 10  ClmDiagnosisCode_2      322357 non-null  object 
 11  ClmDiagnosisCode_3      203257 non-null  object 
 12  ClmDiagnosisCode_4      125596 non-null  object 
 13  ClmDiagnosisCode_5      74344 non-null   object 
 14  ClmDiagnosisCode_6  

<div class="alert alert-block alert-info"> <strong>Outpatient</strong> dataframe consists of 27 columns(features)
</div>

***
# `Merge inpatient & outpatient dataframes`

In [7]:
#merge two dataframes 
frames=[outpatient_df, inpatient_df]
outin_patient = pd.concat(frames)

# view first few records
outin_patient.head()

BeneID    ClaimID ClaimStartDt  ClaimEndDt  Provider  \
0  BENE11002  CLM624349   2009-10-11  2009-10-11  PRV56011   
1  BENE11003  CLM189947   2009-02-12  2009-02-12  PRV57610   
2  BENE11003  CLM438021   2009-06-27  2009-06-27  PRV57595   
3  BENE11004  CLM121801   2009-01-06  2009-01-06  PRV56011   
4  BENE11004  CLM150998   2009-01-22  2009-01-22  PRV56011   

   InscClaimAmtReimbursed AttendingPhysician OperatingPhysician  \
0                      30          PHY326117                NaN   
1                      80          PHY362868                NaN   
2                      10          PHY328821                NaN   
3                      40          PHY334319                NaN   
4                     200          PHY403831                NaN   

  OtherPhysician ClmDiagnosisCode_1  ... ClmProcedureCode_2  \
0            NaN              78943  ...                NaN   
1            NaN               6115  ...                NaN   
2            NaN               2723  ...                NaN   
3            NaN              71988  ...                NaN   
4            NaN              82382  ...                NaN   

  ClmProcedureCode_3 ClmProcedureCode_4 ClmProcedureCode_5 ClmProcedureCode_6  \
0                NaN                NaN                NaN                NaN   
1                NaN                NaN                NaN                NaN   
2                NaN                NaN                NaN                NaN   
3                NaN                NaN                NaN                NaN   
4                NaN                NaN                NaN                NaN   

  DeductibleAmtPaid ClmAdmitDiagnosisCode AdmissionDt DischargeDt  \
0               0.0                 56409         NaN         NaN   
1               0.0                 79380         NaN         NaN   
2               0.0                   NaN         NaN         NaN   
3               0.0                   NaN         NaN         NaN   
4               0.0                 71947         NaN         NaN   

   DiagnosisGroupCode  
0                 NaN  
1                 NaN  
2                 NaN  
3                 NaN  
4                 NaN  

[5 rows x 30 columns]

In [8]:
outin_patient.to_csv('outin_patient.csv')

***
# `Investigating the in&out patients dataset`

In [9]:
outin_patient.shape

(558211, 30)

#### Checking the percentage of the NaNs

In [10]:
def check_missing(df):
    # Check the percentage of Nan in dataset
    total = df.isnull().sum().sort_values(ascending=False)
    percent = (df.isnull().sum()/len(df)).sort_values(ascending=False)
    missing_data = pd.concat([total, percent], axis=1, keys=['Total', 'Percent'])
    return missing_data 

In [11]:
check_missing(outin_patient)

Total   Percent
ClmProcedureCode_6      558211  1.000000
ClmProcedureCode_5      558202  0.999984
ClmProcedureCode_4      558093  0.999789
ClmProcedureCode_3      557242  0.998264
ClmDiagnosisCode_10     553201  0.991025
ClmProcedureCode_2      552721  0.990165
ClmProcedureCode_1      534901  0.958242
DiagnosisGroupCode      517737  0.927493
DischargeDt             517737  0.927493
AdmissionDt             517737  0.927493
ClmDiagnosisCode_9      516396  0.925091
ClmDiagnosisCode_8      504767  0.904258
ClmDiagnosisCode_7      492034  0.881448
ClmDiagnosisCode_6      473819  0.848817
ClmDiagnosisCode_5      446287  0.799495
OperatingPhysician      443764  0.794975
ClmAdmitDiagnosisCode   412312  0.738631
ClmDiagnosisCode_4      393675  0.705244
OtherPhysician          358475  0.642185
ClmDiagnosisCode_3      315156  0.564582
ClmDiagnosisCode_2      195606  0.350416
ClmDiagnosisCode_1       10453  0.018726
AttendingPhysician        1508  0.002701
DeductibleAmtPaid          899  0.001611
ClaimID                      0  0.000000
InscClaimAmtReimbursed       0  0.000000
Provider                     0  0.000000
ClaimEndDt                   0  0.000000
ClaimStartDt                 0  0.000000
BeneID                       0  0.000000

In [12]:
def NAN_50(df):
    """
    get features have more than 50% missing values
    args: dataframe
    return : features have missing values more than 50%
    """ 
    NaN_50 = df.isna().sum()/len(df)*100 > 50 
    return NaN_50[NaN_50 == True]

In [13]:
#get outin_patient features have more than 50% missing
NAN_50(outin_patient)

OperatingPhysician       True
OtherPhysician           True
ClmDiagnosisCode_3       True
ClmDiagnosisCode_4       True
ClmDiagnosisCode_5       True
ClmDiagnosisCode_6       True
ClmDiagnosisCode_7       True
ClmDiagnosisCode_8       True
ClmDiagnosisCode_9       True
ClmDiagnosisCode_10      True
ClmProcedureCode_1       True
ClmProcedureCode_2       True
ClmProcedureCode_3       True
ClmProcedureCode_4       True
ClmProcedureCode_5       True
ClmProcedureCode_6       True
ClmAdmitDiagnosisCode    True
AdmissionDt              True
DischargeDt              True
DiagnosisGroupCode       True
dtype: bool

*** 

### `BeneId`: Beneficiary Patient ID

In [14]:
#get the BeneID dtype
outin_patient['BeneID'].dtype

dtype('O')

In [15]:
outin_patient['BeneID'].unique()

array(['BENE11002', 'BENE11003', 'BENE11004', ..., 'BENE159064',
       'BENE159103', 'BENE159175'], dtype=object)

<div class="alert alert-block alert-info">Sounds all <strong>BeneID</strong> values start with <strong>BENE</strong>. To make sure of that, we will count the occurrences  of BenID start with this text 
</div>

In [16]:
outin_patient['BeneID'].str.startswith('BENE').value_counts().values[0] == outin_patient.shape[0]

True

<div class="alert alert-block alert-success">All the <strong>BeneID</strong> values are valid and there is no missing values. In Addition, our assumption is true (all BeneID start with BENE)
</div>

***

### `ClaimID`: Claim ID

In [17]:
outin_patient['ClaimID'].dtype

dtype('O')

In [18]:
outin_patient['ClaimID'].unique()

array(['CLM624349', 'CLM189947', 'CLM438021', ..., 'CLM76485', 'CLM79949',
       'CLM69948'], dtype=object)

<div class="alert alert-block alert-info">Sounds all <strong>ClaimID</strong> values start with <strong>CLM</strong>. To make sure of that, we will count the occurrences  of ClaimID start with this text 
</div>

In [19]:
outin_patient['ClaimID'].str.startswith('CLM').value_counts().values[0] == outin_patient.shape[0]

True

<div class="alert alert-block alert-success">All the <strong>ClaimID</strong> values are valid and there is no missing values. In Addition, our assumption is true (all ClaimID start with CLM)
</div>

***

### `ClaimStartDt`: Claim Start Date

In [20]:
outin_patient['ClaimStartDt'].dtype

dtype('O')

<div class="alert alert-block alert-warning">The date datatype is an object.

In [21]:
#converting the date object into datetime
outin_patient['ClaimStartDt'] = pd.to_datetime(outin_patient['ClaimStartDt'])
outin_patient['ClaimStartDt'].dtype

dtype('<M8[ns]')

<div class="alert alert-block alert-success">All the <strong>ClaimStartDt</strong> values are valid and there is no missing values.
</div>

***

### `ClaimEndDt`: Claim Start Date

In [22]:
outin_patient['ClaimEndDt'].dtype

dtype('O')


<div class="alert alert-block alert-warning">The date datatype is an object.

In [23]:
#converting the date object into datetime
outin_patient['ClaimEndDt'] = pd.to_datetime(outin_patient['ClaimEndDt'])
outin_patient['ClaimEndDt'].dtype


dtype('<M8[ns]')


<div class="alert alert-block alert-success">All the <strong>ClaimEndDt</strong> values are valid and there is no missing values.
</div>

***

### `AdmissionDt`


In [24]:
outin_patient['AdmissionDt'].dtype

dtype('O')

In [25]:
#converting the date object into datetime
outin_patient['AdmissionDt'] = pd.to_datetime(outin_patient['AdmissionDt'])
outin_patient['AdmissionDt'].dtype

dtype('<M8[ns]')


<div class="alert alert-block alert-info">All the <strong>AdmissionDt</strong> values are valid and there is 517737 missing values(0.927493). NOTE : inpatient has no missing values in AdmissionDt. All missing values come from outpatient since there is no patient has AdmissionDt
</div>

***

### `DischargeDt`


In [26]:
outin_patient['DischargeDt'].dtype

dtype('O')

In [27]:
#converting the date object into datetime
outin_patient['DischargeDt'] = pd.to_datetime(outin_patient['DischargeDt'])
outin_patient['DischargeDt'].dtype

dtype('<M8[ns]')


<div class="alert alert-block alert-info">All the <strong>DischargeDt</strong> values are valid and there is 517737 missing values(0.927493). NOTE : inpatient has no missing values in DischargeDt. All missing values come from outpatient since there is no patient has DischargeDt
</div>

***

### `DiagnosisGroupCode`

In [28]:
outin_patient['DiagnosisGroupCode'].dtype

dtype('O')

In [29]:
outin_patient.DiagnosisGroupCode.value_counts()

882    179
884    174
166    168
183    165
939    163
      ... 
765      1
768      1
775      1
927      1
955      1
Name: DiagnosisGroupCode, Length: 736, dtype: int64


<div class="alert alert-block alert-info"> <strong>DiagnosisGroupCode</strong> have 517737 missing values(0.927493). 
    <ul>
        <li>NOTE : inpatient has no missing values in DiagnosisGroupCode. All missing values come from outpatient since there is no patient has DiagnosisGroupCode</li>
        <li> NOTE : DiagnosisGroupCode should have<strong> int </strong> type, so in the next level we have to convert it into <strong>int</strong> type </li>
        </ul>
</div>

***

### `Provider`

In [30]:
outin_patient['Provider'].dtype

dtype('O')

In [31]:
outin_patient.Provider.value_counts()

PRV51459    8240
PRV53797    4739
PRV51574    4444
PRV53918    3588
PRV54895    3436
            ... 
PRV54503       1
PRV57386       1
PRV55787       1
PRV53643       1
PRV52443       1
Name: Provider, Length: 5410, dtype: int64

In [32]:
outin_patient['Provider'].str.startswith('PRV').value_counts().values[0] == outin_patient.shape[0]

True

<div class="alert alert-block alert-success">All the <strong>Provider</strong> values are valid and there is no missing values. In Addition, all <strong>Provider</strong> values start with <strong>PRV</strong> </div>
    

***

### `InscClaimAmtReimbursed` : Insurance Claim Amount Reimbursed

In [33]:
outin_patient['InscClaimAmtReimbursed'].dtype

dtype('int64')

In [34]:
#check if there is an abnormal values.
print("The minimum value of InscClaimAmtReimbursed -> ", outin_patient['InscClaimAmtReimbursed'].min(),
      "\n The maximum value of InscClaimAmtReimbursed -> ", outin_patient['InscClaimAmtReimbursed'].max())

The minimum value of InscClaimAmtReimbursed ->  0 
 The maximum value of InscClaimAmtReimbursed ->  125000


<div class="alert alert-block alert-success">All the <strong>InscClaimAmtReimbursed</strong> values are valid and there is no missing values.
</div>

***

### `AttendingPhysician, OperatingPhysician, OtherPhysician` 

In [35]:
#create new dataframe for Physician
PhysicianList_ = ['AttendingPhysician', 'OperatingPhysician', 'OtherPhysician' ]
PhysicianList = outin_patient[PhysicianList_]

#get Physicians datatype
PhysicianList.dtypes

AttendingPhysician    object
OperatingPhysician    object
OtherPhysician        object
dtype: object

In [36]:
# check missing values for Physicians
check_missing(PhysicianList)

Total   Percent
OperatingPhysician  443764  0.794975
OtherPhysician      358475  0.642185
AttendingPhysician    1508  0.002701

In [37]:
# Iterate over the sequence of column 
for column in PhysicianList:
   # Select column by column name using [] operator
   columnSeriesObj = PhysicianList[column]
    
    #print unique values per column
   print(column, PhysicianList[column].unique(),'\n')

   #print the occurrences of PhysicianType that start with PHY
   print('The number of',column,'start with PHY',PhysicianList[column].str.startswith('PHY').value_counts(),'\n')
   print('----------------')

AttendingPhysician ['PHY326117' 'PHY362868' 'PHY328821' ... 'PHY411654' 'PHY316716'
 'PHY393635'] 

The number of AttendingPhysician start with PHY True    556703
Name: AttendingPhysician, dtype: int64 

----------------
OperatingPhysician [nan 'PHY366436' 'PHY359916' ... 'PHY420356' 'PHY332778' 'PHY430902'] 

The number of OperatingPhysician start with PHY True    114447
Name: OperatingPhysician, dtype: int64 

----------------
OtherPhysician [nan 'PHY346833' 'PHY311407' ... 'PHY383595' 'PHY431760' 'PHY398228'] 

The number of OtherPhysician start with PHY True    199736
Name: OtherPhysician, dtype: int64 

----------------


<div class="alert alert-block alert-warning"> <strong>AttendingPhysician</strong> have 1508 missing values (0.002701) & the values format are valid. In Addition, all <strong>AttendingPhysician</strong>  starts with <strong> PHY </strong>
</div>

<div class="alert alert-block alert-warning"> <strong>OperatingPhysician</strong> have 443764 missing values (0.794975) & the values format are valid. In Addition, all <strong>OperatingPhysician</strong>  starts with <strong> PHY </strong>
</div>

<div class="alert alert-block alert-warning"> <strong>OtherPhysician</strong> have 358475 missing values (0.642185) & the values format are valid. In Addition, all <strong>OtherPhysician</strong>  starts with <strong> PHY </strong>
</div>

***

### `DeductibleAmtPaid` : Deductible Amount Paid

In [38]:
outin_patient['DeductibleAmtPaid'].dtype

dtype('float64')

In [39]:
#check if there is an abnormal values.
print("The minimum value of Deductible Amount Paid -> ", outin_patient['DeductibleAmtPaid'].min(),
      "\n The maximum value Deductible Amount Paid-> ", outin_patient['DeductibleAmtPaid'].max())

The minimum value of Deductible Amount Paid ->  0.0 
 The maximum value Deductible Amount Paid->  1068.0


<div class="alert alert-block alert-success">All the <strong>ClaimStartDt</strong> values are valid and there is 899 missing values (0.001611).
</div>

***


### `ClmAdmitDiagnosisCode` : Claim Admission Diagnosis Code

In [40]:
outin_patient['ClmAdmitDiagnosisCode'].dtype

dtype('O')

In [41]:
outin_patient['ClmAdmitDiagnosisCode'].unique()

array(['56409', '79380', nan, ..., '5581', '20282', '41061'], dtype=object)

<div class="alert alert-block alert-warning"> <strong>ClmAdmitDiagnosisCode</strong> have 412312 missing values (0.738631) and it's code contains text,numbers ..
</div>

***

### `ClmDiagnosisCode`:  Claim Diagnosis Code

In [42]:
# create list have all ClmDiagnosisCode version
ClmDiagnosisCodeList_ = ['ClmDiagnosisCode_1', 'ClmDiagnosisCode_2', 'ClmDiagnosisCode_3',
                         'ClmDiagnosisCode_4', 'ClmDiagnosisCode_5', 'ClmDiagnosisCode_6',      
                         'ClmDiagnosisCode_7', 'ClmDiagnosisCode_8', 'ClmDiagnosisCode_9',     
                         'ClmDiagnosisCode_10']

In [43]:
# create new dataframe for ClmDiagnosisCode
ClmDiagnosisCodeList = outin_patient[ClmDiagnosisCodeList_]

#get datatypes of ClmDiagnosisCode version
ClmDiagnosisCodeList.dtypes

ClmDiagnosisCode_1     object
ClmDiagnosisCode_2     object
ClmDiagnosisCode_3     object
ClmDiagnosisCode_4     object
ClmDiagnosisCode_5     object
ClmDiagnosisCode_6     object
ClmDiagnosisCode_7     object
ClmDiagnosisCode_8     object
ClmDiagnosisCode_9     object
ClmDiagnosisCode_10    object
dtype: object

In [44]:
#check missing in ClmDiagnosisCode versions
check_missing(ClmDiagnosisCodeList)

Total   Percent
ClmDiagnosisCode_10  553201  0.991025
ClmDiagnosisCode_9   516396  0.925091
ClmDiagnosisCode_8   504767  0.904258
ClmDiagnosisCode_7   492034  0.881448
ClmDiagnosisCode_6   473819  0.848817
ClmDiagnosisCode_5   446287  0.799495
ClmDiagnosisCode_4   393675  0.705244
ClmDiagnosisCode_3   315156  0.564582
ClmDiagnosisCode_2   195606  0.350416
ClmDiagnosisCode_1    10453  0.018726

In [45]:
# Iterate over the sequence of column 
for column in ClmDiagnosisCodeList:
   # Select column by column name using [] operator
   columnSeriesObj = ClmDiagnosisCodeList[column]
    
    #print unique values per column
   print(column, ClmDiagnosisCodeList[column].unique(),'\n')



ClmDiagnosisCode_1 ['78943' '6115' '2723' ... '6172' '65661' '85145'] 

ClmDiagnosisCode_2 ['V5866' nan '30000' ... '73004' '55012' 'V6111'] 

ClmDiagnosisCode_3 ['V1272' nan '72887' ... 'E8529' 'E9504' '27786'] 

ClmDiagnosisCode_4 [nan '4280' 'V854' ... '1175' '94512' '7313'] 

ClmDiagnosisCode_5 [nan '7197' '72751' ... '2510' '28861' '99678'] 

ClmDiagnosisCode_6 [nan 'V4577' '4019' ... '73609' '34570' '5173'] 

ClmDiagnosisCode_7 [nan '9597' 'V5869' ... '3562' '40501' '2212'] 

ClmDiagnosisCode_8 [nan '8449' '78659' ... '6079' '25093' '95203'] 

ClmDiagnosisCode_9 [nan '71596' '34830' ... '29534' 'V0261' '36966'] 

ClmDiagnosisCode_10 [nan 'V059' 'V4581' ... '56981' '68110' 'V5863'] 



<div class="alert alert-block alert-warning"> 
    <ul>
        <li>All of <strong>ClmDiagnosisCode</strong> have object datatype and it's code contains text,numbers...</li>
        <li><strong>ClmDiagnosisCode</strong> version 1&2 have missing values less than 50%</li>
        <li><strong>ClmDiagnosisCode</strong> version from 3 to 10  have missing values more than 50%</li>
    </ul>
    
</div>

***

### `ClmProcedureCode`:  Claim Procedure Code

In [46]:
# create list have all ClmProcedureCode version
ClmProcedureCode_ = ['ClmProcedureCode_1', 'ClmProcedureCode_2',
                    'ClmProcedureCode_3', 'ClmProcedureCode_4',
                    'ClmProcedureCode_5', 'ClmProcedureCode_6']     

In [47]:
# create new dataframe for ClmProcedureCode
ClmProcedureCode = outin_patient[ClmProcedureCode_]

#get datatypes of ClmProcedureCode version
ClmProcedureCode.dtypes

ClmProcedureCode_1    float64
ClmProcedureCode_2    float64
ClmProcedureCode_3    float64
ClmProcedureCode_4    float64
ClmProcedureCode_5    float64
ClmProcedureCode_6    float64
dtype: object

In [48]:
#check missing in ClmProcedureCode versions
check_missing(ClmProcedureCode)

Total   Percent
ClmProcedureCode_6  558211  1.000000
ClmProcedureCode_5  558202  0.999984
ClmProcedureCode_4  558093  0.999789
ClmProcedureCode_3  557242  0.998264
ClmProcedureCode_2  552721  0.990165
ClmProcedureCode_1  534901  0.958242

In [49]:
# Iterate over the sequence of column 
for column in ClmProcedureCode:
   # Select column by column name using [] operator
   columnSeriesObj = ClmProcedureCode[column]
    
    #print unique values per column
   print(column, ClmProcedureCode[column].unique(),'\n')

ClmProcedureCode_1 [  nan 9672. 4573. ... 9815. 9351. 9938.] 

ClmProcedureCode_2 [  nan 3811. 7820. 4439. 4019. 1741.  496. 4571. 9982. 6929. 5849. 4142.
 2724. 7840. 2763. 2731. 7862. 5119. 7804. 9971. 5739. 5733.  412. 4139.
 7904. 2762.  311. 7876. 5185. 5845. 4210. 3570. 5198. 7802. 9975. 9092.
 7830. 4441.  570. 5781. 6141. 4414. 9972. 1122. 4264. 9986. 1629. 2939.
 2749.   42.  260. 5718. 2761. 9973. 4422. 7850. 3320. 5641. 4011. 7907.
 6829. 7906. 5305. 9961. 2819.  539. 1749.  500.  185. 7851. 4254. 3485.
 2752. 7837. 8026. 3331. 9974. 4263. 1733. 4465. 7994. 7845. 2764. 6071.
  920. 7823. 3963. 8670. 8072. 3572. 3542. 2753. 9052. 3812. 6823.  193.
 7859. 5521. 5789. 8054. 4821.  780. 3840. 3962. 7842. 7854. 4240. 3371.
 4241. 9072. 1623. 4109. 9054. 7856. 1519. 5712. 5369. 3842. 8921. 4561.
 7921. 2722. 3844. 6212. 4292. 5184. 3481. 4104. 8881. 6191. 3899. 5693.
 7812. 4464. 5695. 8404. 7231. 7843. 2630. 8052. 2251. 3009. 5679. 1453.
 4242. 4552. 7993. 5569. 5363. 5183. 7852.

<div class="alert alert-block alert-warning"> 
    <ul>
        <li>All of <strong>ClmProcedureCode</strong> have float64 datatype </li>
        <li> All values are missing in <strong>ClmProcedureCode_6</strong> </li>
        <li><strong>ClmProcedureCode</strong> version from  <strong> 1 to 5  </strong>have about 90% missing values </li>
    </ul>
    
</div>

***